packages/ui/src/components/drawer/DrawerHeader.tsx

```diff
@@ -0,0 +1,66 @@
+import { Children, HTMLAttributes, ReactNode } from 'react'
+
+import { Icon, IconProps, Logo, LogoProps } from '@/components'
+import { cn, getComponentDisplayName } from '@/utils'
+import { Drawer as DrawerPrimitive } from 'vaul'
+
+import { DrawerTagline } from './Drawer.Tagline'
+
+type DrawerHeaderBaseProps = Omit<HTMLAttributes<HTMLDivElement>, 'children'> & {
+  children: ReactNode
+}
+
+type DrawerHeaderIconOnlyProps = {
+  icon: IconProps['name']
+  logo?: never
+}
+
+type DrawerHeaderLogoOnlyProps = {
+  logo: LogoProps['name']
+  icon?: never
+}
+
+type DrawerHeaderNoIconOrLogoProps = {
+  icon?: never
+  logo?: never
+}
+
+export type DrawerHeaderProps = DrawerHeaderBaseProps &
+  (DrawerHeaderIconOnlyProps | DrawerHeaderLogoOnlyProps | DrawerHeaderNoIconOrLogoProps)
```

Comment:
This stuff is a pain in the ass with props on a component. _Much_ easier with arguments on a standard function. Makes a big difference to the dev using the component though 👍🏻